src/main/groovy/edu/oregonstate/mist/directoryapi/DirectoryEntityDAO.groovy

```diff
@@ -190,6 +190,23 @@ class DirectoryEntityDAO {
     }
 
     /**
+     * Gets an osuuid as a long
+     *
+     * @param ldapEntry
+     * @return Long osuuid
+     */
+    private static Long getOSUUid(LdapEntry ldapEntry) {
+        String osuuidString = get(ldapEntry, 'osuuid')
+        Long osuuid
+
+        if (osuuidString) {
```

Comment:
Let's check and make sure that osuuidString is a numeric value before we call the parseLong value.